### PR TITLE
Kotlin 1.7.0 prep

### PR DESCRIPTION
**Pull request by ZacSweers**
- Use the new kotlin base plugin as a hook
- Add a toe-hold for supporting the K2 compiler via property
- Clean up redundant tasks
- Update deps along the way
- Try and fail to use `-xjdk-release`
  - This has confusing behavior with jvmTarget and toolchains
- Remove redundant RequiresOptIn arg
  - Add it back because https://youtrack.jetbrains.com/issue/KT-52720

### CODE REVIEW

File: slack-plugin/src/main/kotlin/slack/gradle/SlackProperties.kt [hunk 1]
```diff
@@ -253,6 +253,13 @@ public class SlackProperties private constructor(private val project: Project) {
   public val kotlinDaemonArgs: String
     get() = stringProperty(KOTLIN_DAEMON_ARGS_KEY, defaultValue = "")
 
+  /**
+   * Flag to enable the new K2 compiler. Plumbed into the kotlinOptions.useK2 property and here to
+   * allow for use from a `local.properties` file.
+   */
+  public val useK2: Boolean
+    get() = booleanProperty("slack.useK2", defaultValue = false)
```
Comment (seankim-android) [inline slack-plugin/src/main/kotlin/slack/gradle/SlackProperties.kt:261]: Nice. It looks like there’s a nice performance boost with this new compiler from https://kotlinlang.org/docs/whatsnew17.html#new-kotlin-k2-compiler-for-the-jvm-in-alpha

File: slack-plugin/src/main/kotlin/slack/gradle/StandardProjectConfigurations.kt [hunk 2]
```diff
@@ -815,43 +837,14 @@ internal class StandardProjectConfigurations {
...
-              file("$buildDir/intermediates/detekt/baseline.xml")
-            } else {
-              file(rootProject.file(baselineFile))
-            }
-        }
-      }
-    }
-
     pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
-      onKotlinPluginApplied()
```
Comment (seankim-android) [inline slack-plugin/src/main/kotlin/slack/gradle/StandardProjectConfigurations.kt:845]: > Use the new kotlin base plugin as a hook

Question for my understanding. Why did we need to invoke this lambda multiple times before, and we don't need it anymore?

**Review by seankim-android — APPROVED** [review]
LGTM

File: slack-plugin/src/main/kotlin/slack/gradle/StandardProjectConfigurations.kt (hunk 2 above)
Comment (ZacSweers) [reply slack-plugin/src/main/kotlin/slack/gradle/StandardProjectConfigurations.kt:845]: before we had to hook in both the jvm and android plugins, but now we get the "base" plugin as a common hook for these common configs. The android plugin has a similar tool

**Pull request merged**